clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -175,6 +176,13 @@ protected AvroGenericDaVinciClient(
     this.managedClients = managedClients;
     this.icProvider = icProvider;
     this.chunkingAdapter = chunkingAdapter;
+
+    this.recordTransformer = daVinciConfig.isRecordTransformerEnabled() ? daVinciConfig.getRecordTransformer(0) : null;
```

Comment:
again, this is kind of sketchy.  Why are we doing a call with version zero?

Reply:
I'm not sure I follow.  Couldn't we just pass it along via configs?

Reply:
When we're creating the [AvroSpecificStoreDeserializerCache](https://github.com/kvargha/venice/blob/14fef3146d21050013ebff53975cea26a48b99f8/clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java#L797-L799), right now we pass in the ValueClass from the RecordTransformer into the constructor. 

In order to access the ValueClass, we have to initialize a RecordTransformer to extract it.

Since `AvroGenericDaVinciClient` doesn't have access to the RecordTransformer that `SIT` creates, I have to create a dummy one here.

[This](https://github.com/linkedin/venice/blob/2cbd14f3d63fcd22b2ddb60460a431ec6cfc62b4/clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformer.java#L267-L273) is the function inside RecordTransformer that returns the ValueClass. Is there a better way to do it?

Reply:
Could we just have the getRecordTransformer also take a schema argument that it gets from the davinciConfig?  That way the user specifies it in the config and then it's accessible at this level.

I really really think that instantiating something that is user defined frivolously is a bad idea because we don't know what the user will have tied to this interface.  It could be very expensive like database instances or file directories.

Reply:
Rereading this I think I'm more and more in favor of that just adding an etnry to the davinci config for the output schema, that way we don't need the member here.

Reply:
I completely agree. Not a big fan of this hack.

In this file we also need the output value class. Since we're no longer going to instantiate the recordTransformer here, this also needs to be added to the DaVinciConfig. Is that ok?

Reply:
ToDo: Create record transformer config that's a subconfig of davinciconfig.